@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1543,8 +1546,11 @@ export class TextElementsRenderer {
         tempScreenPosition.x = tempPoiScreenPosition.x = screenPosition.x;
         tempScreenPosition.y = tempPoiScreenPosition.y = screenPosition.y;
 
-        // Scale the text depending on the label's distance to the camera.
-        const textDistance = this.m_viewState.worldCenter.distanceTo(position);
+        // Scale the text depending on the label's distance to the camera "zero" plane - near
+        // plane at zero distance. The result is the same as taking the distance from "Z"
+        // coordinate in camera space. This way we avoid co-planar labels' scale variation.
+        const labelCamVec = this.m_tmpLabelCamVec.copy(position).sub(this.m_viewState.worldCenter);
```

Comment:
This should be done in computeViewDistance function in Placement.ts. That function should be call from here, as it is done for path labels in addPathLabel. Check that the calculations for path labels makes sense too.